docs/template_plugin/src/template_executable_network.cpp

```diff
@@ -112,10 +112,12 @@ void TemplatePlugin::ExecutableNetwork::CompileNetwork(const std::shared_ptr<con
 
     // Generate backend specific blob mappings. For example Inference Engine uses not ngraph::Result nodes friendly name
     // as inference request output names but the name of the layer before.
+    size_t idx = 0;
     for (auto&& result : _function->get_results()) {
         const auto& input = result->input_value(0);
         auto name = ngraph::op::util::get_ie_output_name(input);
-        _outputIndex.emplace(name, _function->get_result_index(result));
+        if (_outputIndex.emplace(name, idx).second)
+            idx++;
```

Comment:
Looks like we don't use this `idx` variable in other places. Do we need it?

Reply:
Previously, in case of duplication, the `name` was assigned to `_function->get_result_index(result)` out of the tensor vector size. After the change, the incrementation of the index is done, when the insertion take place to match the tensor vector index. Could you point what other places you have in mind?